### DISCUSSION TITLE
What is the utility of SAMTOOLS_MERGE ?

### DISCUSSION THREAD

**Discussion by abhi18av** [TORCH-Consortium/MAGMA#26] [Q&A] [answered]
Hello everyone,

I was wondering whether you could help me understand what's the use-case for `samtools merge` in the `CALL_WF` workflow 

https://github.com/abhi18av/xbs-nf/blob/07c8b73e037f89a2f821ff6d4e935e91b05d3ea8/workflows/call_wf.nf#L35

From the original script XBS_call.sh (Line-31), it seems like, this step would expect a bunch of BAM files for a particular sample - however I'm not sure why would a single sample have multiple bam files?

**Comment by LennertVerboven** [comment, marked as answer]
Hi


This has to do with samples for which the first sequencing run did not produce enough sequence data. Such a sample can be resequenced. The mapped bam files for both sequencing runs are then combined during the samtools merge. This is an edge case and I dont think it has happened before. Which data need to merged is decided based on the names of the the bam files.


Tim correct me if i am wrong.


Kind regards


Lennert Verboven



[logo_uantwerp] <https://www.uantwerp.be>    [logo_torch]  <https://torch-consortium.com/>

        TORCH - Tuberculosis Omics ResearCH consortium
Global Health Institute
Gouverneur Kinsbergencentrum - Room 1.16
Doornstraat 331 - 2610 Wilrijk - Belgium
***@***.*** ***@***.***>

________________________________
From: Abhinav Sharma ***@***.***>
Sent: 26 October 2021 13:05:55
To: abhi18av/xbs-nf
Cc: Subscribed
Subject: [abhi18av/xbs-nf] What is the utility of SAMTOOLS_MERGE ? (Discussion #26)


CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.


Hello everyone,

I was wondering whether you could help me understand what's the use-case for samtools merge in the CALL_WF workflow

https://github.com/abhi18av/xbs-nf/blob/07c8b73e037f89a2f821ff6d4e935e91b05d3ea8/workflows/call_wf.nf#L35<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fabhi18av%2Fxbs-nf%2Fblob%2F07c8b73e037f89a2f821ff6d4e935e91b05d3ea8%2Fworkflows%2Fcall_wf.nf%23L35&data=04%7C01%7Clennert.verboven%40uantwerpen.be%7C26880ee4e55445c66a4108d998709604%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637708431600188682%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=MeLUzXqRKPjWhDF8esKTMO1bxQbY0LfdzaRkmP2XORI%3D&reserved=0>

From the original script XBS_call.sh (Line-31), it seems like, this step would expect a bunch of BAM files for a particular sample - however I'm not sure why would a single sample have multiple bam files?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fabhi18av%2Fxbs-nf%2Fdiscussions%2F26&data=04%7C01%7Clennert.verboven%40uantwerpen.be%7C26880ee4e55445c66a4108d998709604%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637708431600188682%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=fCF2aG7fjwtzDt5lAJ%2Buc95C3X0ZhpPDwZGF9BKIjMc%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACWGSFOAZ2BCBX6UA3GO33LUI2DRHANCNFSM5GXQI2YQ&data=04%7C01%7Clennert.verboven%40uantwerpen.be%7C26880ee4e55445c66a4108d998709604%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637708431600198645%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=vPQdkIiKpvrYkw3A7RiIYyiQiAzTY5wPS28zG2DMvwA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Clennert.verboven%40uantwerpen.be%7C26880ee4e55445c66a4108d998709604%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637708431600198645%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Ur7eCdosftJbpT7owvc0WMmn3dU43Zn17ek1ZhB5yr0%3D&reserved=0> or Android<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Clennert.verboven%40uantwerpen.be%7C26880ee4e55445c66a4108d998709604%7C792e08fb2d544a8eaf72202548136ef6%7C0%7C0%7C637708431600208594%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=INymBFSoCARWmlXkg%2BjmAAFmxJRQzOTet0zwma428uY%3D&reserved=0>.

**Reply by abhi18av** [reply]
Thanks @LennertVerboven  for the update - in that case, I'll keep this code commented and would skip this step.

In future, if there's a need to combine multiple BAMs for a sample, then it should be straight-forward to adapt this code.

**Comment by TimHHH** [comment]
Lennert is correct, this is for merging multiple libraries from the same sample. This may happen when one library is not too good and a better one also available, which happens some times, but this can also happen when we decide we simply want more/deeper data for a sample, this also happens some times. And there are more scenarios. It is important enough that this step should be implemented now please, we will encounter this issue in the dataset for the paper. Please see my post in the testing section on the impact of library/sample names.
cheers.

**Reply by abhi18av** [reply]
Sure, @TimHHH  - this makes it clearer to me. [The code is here](https://github.com/abhi18av/xbs-nf/blob/90282ecb7089f45419d014bc384e726c09fc7961/workflows/call_wf.nf#L38) of course, but I think that the design might need to be update to accomodate this consideration.

I'll track this as a task as well for implementing later.
